jkube-kit/generator/api/src/main/java/org/eclipse/jkube/generator/api/support/BaseGenerator.java

```diff
@@ -289,4 +294,33 @@ protected void addSchemaLabels(BuildConfiguration.BuildConfigurationBuilder buil
             buildBuilder.labels(labels);
         }
     }
-}
+
+    public void checkAndWarnIfProjectHasNotBeenBuilt() {
+      try {
+        File finalOutputArtifact = JKubeProjectUtil.getFinalOutputArtifact(getProject());
+        if (finalOutputArtifact == null) {
+          log.error(
+              "Final output artifact file was not detected. The project may have not been built. " +
+                  "HINT: try to compile and package your application prior to running the container image build task.");
+          return;
+        }
+
+        Path lastModifiedTimeSavePath = getProject().getBuildDirectory().toPath();
+
+        long currentLastModifiedTime = finalOutputArtifact.lastModified();
+        long previousLastModifiedTime = retrievePreviousArtifactLastModifiedTime(lastModifiedTimeSavePath);
+
+        boolean isCurrentFinalArtifactSameAsPrevious = currentLastModifiedTime == previousLastModifiedTime;
```

Comment:
I think this would make plugin print warning always after first `k8s:build` run. Do we really want this behavior?

Reply:
> I think this would make plugin print warning always after first `k8s:build` run. Do we really want this behavior?

but we don't get any warning after first k8s:build. Warning is issued only if curentLastModifiedTime is equal to previousLastModifiedTime. On the first run, this will be false.